aggregator/src/aggregator.rs

```diff
@@ -2751,375 +2647,481 @@ where
+        Self(
+            meter
+                .u64_counter("janus_aggregator_responses_total")
+                .with_description(
+                    "Count of requests handled by the aggregator, by method, route, and response status.",
+                )
+                .init(),
+        )
+    }
+}
 
-/// CollectionId is 16 bytes, hence 22 bytes of Base64URL
-const COLLECTION_JOB_PATH_REGEX: &str = "collection_jobs/[A-Za-z0-9-_]{22}";
+#[async_trait]
+impl Handler for StatusCounter {
+    async fn run(&self, conn: Conn) -> Conn {
+        conn
+    }
 
-/// Constructs a Warp filter with endpoints common to all aggregators.
-pub fn aggregator_filter<C: Clock>(
+    async fn before_send(&self, conn: Conn) -> Conn {
+        // Check for the error code set by the Error handler implementation.
+        let error_code_opt = conn.state::<ErrorCode>().map(|error_code| error_code.0);
+        let error_code = if let Some(status) = conn.status() {
+            if status.is_client_error() || status.is_server_error() {
+                error_code_opt.unwrap_or("unknown")
+            } else {
+                // Set the label to an empty string on success.
+                ""
+            }
+        } else {
+            // No status is set, it will fall back to 404.
+            error_code_opt.unwrap_or("unknown")
+        };
+        // Fetch the method.
+        let method = conn.method().as_str();
+        // Check for the route set by the router.
+        let route = conn
+            .route()
+            .map(ToString::to_string)
+            .unwrap_or_else(|| "unknown".to_owned());
+        self.0.add(
+            &Context::current(),
+            1,
+            &[
+                KeyValue::new("method", method),
+                KeyValue::new("route", route),
+                KeyValue::new("error_code", error_code),
+            ],
+        );
+        conn
+    }
+}
+
+/// Constructs a Trillium handler for the aggregator.
+pub fn aggregator_handler<C: Clock>(
     datastore: Arc<Datastore<C>>,
     clock: C,
     cfg: Config,
-) -> Result<BoxedFilter<(impl Reply,)>, Error> {
+) -> Result<impl Handler, Error> {
     let meter = opentelemetry::global::meter("janus_aggregator");
-    let response_time_histogram = meter
-        .f64_histogram("janus_aggregator_response_time")
-        .with_description("Elapsed time handling incoming requests, by endpoint & status.")
-        .with_unit(Unit::new("seconds"))
-        .init();
+    let aggregator = Arc::new(Aggregator::new(datastore, clock, &meter, cfg));
+
+    Ok((
+        State(aggregator),
+        metrics("janus_aggregator"),
+        Router::new()
+            .get("hpke_config", instrumented(api(hpke_config::<C>)))
+            .with_route(
+                trillium::Method::Options,
+                "hpke_config",
+                hpke_config_cors_preflight,
+            )
+            .put("tasks/:task_id/reports", instrumented(api(upload::<C>)))
+            .with_route(
```

Comment:
The router has default options handling that we may want to disable: `Router::new().without_options_handling().get…` [`Router::without_options_handling`](https://docs.rs/trillium-router/latest/trillium_router/struct.Router.html#method.without_options_handling)